lib/compute/index.js

```diff
@@ -50,17 +64,17 @@ exports.image = function image(id, payload) {
         return reject(err);
       }
       var buf = Buffer.concat(arr);
-      imageCc.enqueue({
-        id: id,
-        payload: buf
-      }, function(err, res) {
-        if (err) {
-          logger.error('process.error', err);
-          reject(AppError.processingError(err));
-        } else {
-          resolve(res);
-        }
-      });
+
+      resolve(P.all(Object.keys(SIZES).map(function(variant) {
+        var hw = SIZES[variant];
+        return enqueue({
+          id: id,
+          suffix: (variant === 'default') ? '' : variant,
+          height: hw.h,
+          width: hw.w,
+          payload: buf
+        });
+      })));
```

Comment:
I dug into the S3 docs, and there is no sort of "transaction" for uploading multiple files, only multiple parts of a single file. It seems to properly accomplish this, we would need to collect any and all errors uploading, and try to delete all images that didn't error. Those delete requests would also need to be monitored, and perhaps retried (up to a maximum?).

*shudder*